File: se321/se321-pz-nikola_tasic_3698/src/main/java/com/example/backend/entity/Tag.java
package com.example.backend.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;
import java.util.List;

/**
 * Blog post tag
 */
@Data
@Entity
@Table(name = "tag")
@EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
public class Tag extends Auditable {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@EqualsAndHashCode.Include
	@Column(name = "tag_id")
	private Integer id;
	@Column(name = "name")
	private String name;
	@ManyToMany(mappedBy = "tags")
	@JsonIgnore
	private List<PostPreview> posts;
}

File: cs322/cs322-pz-nikola_tasic_3698/blog-backend/Data/Entity/Tag.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace blog_backend.Data;

[Table("tag")]
public class Tag : Auditable {
	[Key, Column("tag_id")]
	public int Id { get; set; }
	[Column("name")]
	public string Name { get; set; }
}